### PR TITLE
Pod egress network shapping Chaos scenario

### PR DESCRIPTION
**Pull request by yogananth-subramanian**
The scenario introduces network latency, packet loss, and bandwidth restriction in the Pod's network interface. The purpose of this scenario is to observe faults caused by random variations in the network.

Below example config applies egress traffic shaping to openshift console.
````
- id: pod_egress_shaping
  config:
    namespace: openshift-console   # Required - Namespace of the pod to which filter need to be applied.
    label_selector: 'component=ui' # Applies traffic shaping to access openshift console.
    network_params:
        latency: 500ms             # Add 500ms latency to egress traffic from the pod.
````

### CODE REVIEW

File: docs/pod_network_shaping.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+### Pod Network shaping
+Scenario to introduce network latency, packet loss, and bandwidth restriction in the Pod's network interface. The purpose of this scenario is to observe faults caused by random variations in the network.
+
+##### Sample scenario config for egress traffic shaping (using plugin)
+'''
+- id: pod_egress_shaping
```
Comment (chaitanyaenr) [inline docs/pod_network_shaping.md:6]: thoughts on adding this - id: pod_egress_shaping as part of pod_network_outage config?

Comment (yogananth-subramanian) [reply docs/pod_network_shaping.md:6]: Yes, that sounds good too.

File: scenarios/plugin.schema.json [hunk 2]
```diff
@@ -2253,6 +2253,166 @@
...
 					"config"
 				]
+			},
+			{
+				"type": "object",
+				"title": "pod_egress_shaping Arcaflow scenarios",
+				"properties": {
+					"id": {
+						"type": "string",
+						"const": "pod_egress_shaping"
```
Comment (paigerube14) [inline scenarios/plugin.schema.json:2263]: should this be pod_network_shaping? Getting error it doesnt exist in the scenario file

Comment (yogananth-subramanian) [reply scenarios/plugin.schema.json:2263]: Hello @paigerube14 it should be "pod_egress_shaping" and vscode is able to populate the required text.
`- id: pod_egress_shaping
  config:
    namespace: 
    test_duration: 90
    wait_duration: 300
    instance_count: 1
    execution_type: parallel`

Comment (paigerube14) [reply scenarios/plugin.schema.json:2263]: Getting error 

"Value must be "pod_network_shaping".yaml-schema: pod_network_shaping Arcaflow scenarios(1)"

**Comment by yogananth-subramanian** [comment]
Hello @chaitanyaenr @paigerube14,  have update the PR, integrating "pod egress shaping"  with "pod network outage", as a single plugin. Your review would be greatly appreciated.

File: scenarios/openshift/pod_network_shaping.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# yaml-language-server: $schema=../plugin.schema.json
+- id: pod_network_shaping
```
Comment (chaitanyaenr) [inline scenarios/openshift/pod_network_shaping.yml:2]: s/pod_network_shaping/pod_egress_shaping

File: docs/pod_network_scenarios.md [hunk 4]
```diff
@@ -1,3 +1,5 @@
+## Pod network Scenarios
```
Comment (chaitanyaenr) [inline docs/pod_network_scenarios.md:1]: Let's add pod network scenarios to the list: https://github.com/redhat-chaos/krkn/tree/main#kubernetesopenshift-chaos-scenarios-supported

**Review by chaitanyaenr — APPROVED** [review]
Ran few tests. LGTM except minor changes ( commented ).

File: docs/pod_network_scenarios.md (hunk 4 above)
Comment (yogananth-subramanian) [reply docs/pod_network_scenarios.md:1]: done.

File: scenarios/openshift/pod_network_shaping.yml (hunk 3 above)
Comment (yogananth-subramanian) [reply scenarios/openshift/pod_network_shaping.yml]: Thanks for the review, fixed in the latest commit.

**Review by chaitanyaenr — APPROVED** [review]
LGTM

**Pull request merged**